### PR TITLE
fix(ws test) deflaky

### PR DESCRIPTION
**Pull request by cirospaciari**
### What does this PR do?

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [X] Code changes

### How did you verify your code works?
Tests

### CODE REVIEW

**Comment by robobun** [comment]
@cirospaciari, your commit 14c204ab8f91d7bcbbdc968aac0661038da27091 is building: [`#1970`](https://buildkite.com/bun/bun/builds/1970)

File: src/bun.js/api/server.zig [hunk 1]
```diff
@@ -4165,7 +4165,8 @@ pub const ServerWebSocket = struct {
         if (this.isClosed() or vm.isShuttingDown())
             return;
 
-        if (handler.onDrain != .zero) {
+        // we should not call drain if we dont have this_value
+        if (handler.onDrain != .zero and !this.this_value.isEmptyOrUndefinedOrNull()) {
```
Comment (Jarred-Sumner) [inline src/bun.js/api/server.zig]: Seems kind of sketchy. Maybe we should call `getThisValue()` instead?

Comment (cirospaciari) [reply src/bun.js/api/server.zig]: Actually Im debugging to see where is the issue but `drain` event is called passing a null value, probably we are unprotecting to soon?

Comment (cirospaciari) [reply src/bun.js/api/server.zig]: Yeah `getThisValue` was not called before because I did not add a open callback, so getThisValue fix this issue

**Pull request merged**